plugins/modules/system/dpkg_divert.py

```diff
@@ -0,0 +1,390 @@
+        to.
+      - The actual package does not have to be installed or even to exist
+        for its name to be valid. If not specified, the diversion is hold
+        by 'LOCAL', that is reserved by/for dpkg for local diversions.
+      - This parameter is ignored when I(state=absent).
+    type: str
+  divert:
+    description:
+      - The location where the versions of file will be diverted.
+      - Default is to add suffix C(.distrib) to the file path.
+      - This parameter is ignored when I(state=absent).
+    type: path
+  rename:
+    description:
+      - Actually move the file aside (when I(state=present)) or back (when
+        I(state=absent)), but only when changing the state of the diversion.
+        This parameter has no effect when attempting to add a diversion that
+        already exists or when removing an unexisting one.
+      - Unless I(force=true), renaming fails if the destination file already
+        exists (this lock being a dpkg-divert feature, and bypassing it being
+        a module feature).
+    type: bool
+    default: no
+  force:
+    description:
+      - When I(rename=true) and I(force=true), renaming is performed even if
+        the target of the renaming exists, i.e. the existing contents of the
+        file at this location will be lost.
+      - This parameter is ignored when I(rename=false).
+    type: bool
+    default: no
+notes:
+  - This module supports I(check_mode).
+requirements:
+  - dpkg-divert >= 1.15.0 (Debian family)
+'''
+
+EXAMPLES = r'''
+- name: Divert /usr/bin/busybox to /usr/bin/busybox.distrib and keep file in place
+  community.general.dpkg_divert:
+    path: /usr/bin/busybox
+
+- name: Divert /usr/bin/busybox by package 'branding'
+  community.general.dpkg_divert:
+    path: /usr/bin/busybox
+    holder: branding
+
+- name: Divert and rename busybox to busybox.dpkg-divert
+  community.general.dpkg_divert:
+    path: /usr/bin/busybox
+    divert: /usr/bin/busybox.dpkg-divert
+    rename: yes
+
+- name: Remove the busybox diversion and move the diverted file back
+  community.general.dpkg_divert:
+    path: /usr/bin/busybox
+    state: absent
+    rename: yes
+    force: yes
+'''
+
+RETURN = r'''
+commands:
+  description: The dpkg-divert commands ran internally by the module.
+  type: list
+  returned: on_success
+  elements: str
+  sample: |-
+    [
+      "/usr/bin/dpkg-divert --no-rename --remove /etc/foobarrc",
+      "/usr/bin/dpkg-divert --package ansible --no-rename --add /etc/foobarrc"
+    ]
+messages:
+  description: The dpkg-divert relevant messages (stdout or stderr).
+  type: list
+  returned: on_success
+  elements: str
+  sample: |-
+    [
+      "Removing 'local diversion of /etc/foobarrc to /etc/foobarrc.distrib'",
+      "Adding 'diversion of /etc/foobarrc to /etc/foobarrc.distrib by ansible'"
+    ]
+diversion:
+  description: The status of the diversion before and after task.
+  type: dict
+  returned: on_success
+  contains:
+    after:
```

Comment:
I'd like to open a discussion for this `before` / `after` returns I've already seen in other modules: My opinion is that 

1. these are more confusing to users then helpful, especially in check mode.
2. there is `--diff` which solves this already for users 

In my opinion, the return should always return the "actual" state.

I would suggest to discuss it and document the decision in a developer docs.

Reply:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#creating-correct-and-informative-module-output states
> Return diff if in diff mode. This is not required for all modules, as it won’t make sense for certain ones, but please include it when applicable.

User documentation for `--diff` mode https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html#showing-differences-with-diff

Reply:
thank you for the point. It's done (I hope correctly):
- the module now supports `diff` mode (also implemented in tests)
- the `diversion` dict refers only to the actual state (or the *state that would be reached*, when running in `check_mode`) and always returned (i.e. in all module.fail_json() and module.exit_json())

Reply:
@resmo  , could you please take a look the diff is implemented as it shoud?